docs/design/api-versions.md

```diff
@@ -32,8 +40,169 @@ For each API spec, generated `AssignPropertiesTo*()` method  will read the value
 
 For each Storage spec, the generated `AssignPropertiesTo*()` and `AssignPropertiesFrom*()` methods will copy the `OriginalVersion` property between versions, preserving the original value.
 
-### API Recovery
-
-Into each storage resource variant, we'll inject a function `OriginalGVK(): GroupVersionKind` which will use `OriginalVersion` to create the GVK required.
+## New conversion interfaces
 
-SequenceDiagram TBC.
+We don't want the overhead of doing a complete conversion of each resource if we're only interested in conversion of the associated `Spec` or `Status`. For example, when submitting a request to ARM we need only do a `Spec` conversion; similiarly, when processing a response from ARM, we need only do a `Status` conversion.
+
+We already have the required infrastructure in the form of the `AssignProperties*()` methods, making it simple to implement these new interfaces.
+
+For conversion of `Spec` types:
+
+``` go
+type ConvertibleSpec interface {
```

Comment:
Not immediately grasping why this helps us since `Spec` already implements `Convertible` that you mention below? Why can't we just use that?

Reply:
Spec's done - and can't implement the standard `Convertible` interface as those methods require conversion to/from the hub resource type. Specs aren't resources, and only resources should implement that one.

Reply:
Doh, I missed that it only applied to top level types somehow. Got it.